packages/cli/generators/service/index.js

```diff
@@ -0,0 +1,242 @@
+      {
+        type: 'input',
+        name: 'name',
+        // capitalization
+        message: utils.toClassName(this.artifactInfo.type) + ' name:',
+        when: this.artifactInfo.name === undefined,
+        validate: utils.validateClassName,
+      },
+    ];
+    return this.prompt(prompts).then(props => {
+      Object.assign(this.artifactInfo, props);
+      return props;
+    });
+  }
+
+  async promptDataSourceName() {
+    if (this.shouldExit()) return false;
+
+    debug('Prompting for a datasource ');
+    let cmdDatasourceName;
+    let datasourcesList;
+
+    // grab the datasourcename from the command line
+    cmdDatasourceName = this.options.datasource
+      ? utils.toClassName(this.options.datasource) + 'Datasource'
+      : '';
+
+    debug(`command line datasource is  ${cmdDatasourceName}`);
+
+    try {
+      datasourcesList = await utils.getArtifactList(
+        this.artifactInfo.datasourcesDir,
+        'datasource',
+        true,
+      );
+      debug(
+        `datasourcesList from ${utils.sourceRootDir}/${
+          utils.datasourcesDir
+        } : ${datasourcesList}`,
+      );
+    } catch (err) {
+      return this.exit(err);
+    }
+
+    const availableDatasources = datasourcesList.filter(item => {
+      debug(
+        `inspecting datasource: ${item} for basemodel: ${VALID_CONNECTORS_FOR_SERVICE}`,
+      );
+      const result = utils.isConnectorOfType(
+        VALID_CONNECTORS_FOR_SERVICE,
+        this.artifactInfo.datasourcesDir,
+        item,
+      );
+      return result;
+    });
+
+    if (availableDatasources.length === 0) {
+      return this.exit(
+        new Error(
+          `${ERROR_NO_DATA_SOURCES_FOUND} ${this.artifactInfo.datasourcesDir}.
+        ${chalk.yellow(
+          'Please visit https://loopback.io/doc/en/lb4/DataSource-generator.html for information on how datasources are discovered',
+        )}`,
+        ),
+      );
+    }
+
+    if (availableDatasources.includes(cmdDatasourceName)) {
+      Object.assign(this.artifactInfo, {
+        dataSourceClass: cmdDatasourceName,
+      });
+    }
+
+    debug(`artifactInfo.dataSourceClass ${this.artifactInfo.dataSourceClass}`);
+
+    return this.prompt([
+      {
+        type: 'list',
+        name: 'dataSourceClass',
+        message: PROMPT_MESSAGE_DATA_SOURCE,
+        choices: availableDatasources,
+        when: !this.artifactInfo.dataSourceClass,
+        default: availableDatasources[0],
+        validate: utils.validateClassName,
+      },
+    ])
+      .then(props => {
+        Object.assign(this.artifactInfo, props);
+        debug(`props after datasource prompt: ${inspect(props)}`);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during datasource prompt: ${err}`);
+        return this.exit(err);
+      });
+  }
+
+  /**
+   * this method will be in charge of inferring and create
+   * the remote interfaces from either SOAP wsdl or REST openApi json
+   * TODO: @marioestradarosa
```

Comment:
What is remaining to do? Could you please add some content to the `TODO` line to explain what needs to be done?